File: LABA5/parserPack/MpaaRatingParser.java
package parserPack;

import elementsPack.MpaaRating;
import exceptionsPack.IncorrectInputException;
import exceptionsPack.NullFieldException;

public class MpaaRatingParser {
    public static MpaaRating parse(String line) throws IncorrectInputException {
        if (line.isEmpty()) return null;
        line = line.toUpperCase().replaceAll("-", "_");
        MpaaRating mpaaRating = line.equals("NC_17") ? MpaaRating.NC_17
                : line.equals("PG") ? MpaaRating.PG
                : line.equals("R") ? MpaaRating.R
                : null;
        if (mpaaRating == null) throw new IncorrectInputException();
        return mpaaRating;
    }
}

File: LABA5kt/src/parserPack/MpaaRatingParser.kt
package parserPack

import elementsPack.MpaaRating
import exceptionsPack.IncorrectInputException
import java.util.*

object MpaaRatingParser {
    /**
     * A method to parse the given string and return the corresponding MpaaRating.
     *
     * @param  line  the string to be parsed
     * @return       the corresponding MpaaRating, or null if the string is empty
     */
    @Throws(IncorrectInputException::class)
    fun parse(line: String): MpaaRating? {
        if (line.replace(" ", "").isEmpty()) return null
        val mpaaRating = when (line.uppercase(Locale.getDefault()).replace("-", "_").replace(" ", "")) {
            "NC_17" -> MpaaRating.NC_17
            "PG" -> MpaaRating.PG
            "R" -> MpaaRating.R
            else -> null
        }
        if (mpaaRating == null) throw IncorrectInputException()
        return mpaaRating
    }
}